language/diem-framework/modules/DiemErrors.move

```diff
@@ -0,0 +1,40 @@
+address 0x1 {
+
+/// Module defining error codes used in Move aborts throughout the framework.
+///
+/// A `u64` error code is constructed from two values:
+///
+///  1. The *error category* which is encoded in the lower 8 bits of the code. Error categories are
+///     declared in this module and are globally unique across the Diem framework. There is a limited
+///     fixed set of predefined categories, and the framework is guaranteed to use those consistently.
+///
+///  2. The *error reason* which is encoded in the remaining 56 bits of the code. The reason is a unique
+///     number relative to the module which raised the error and can be used to obtain more information about
+///     the error at hand. It is mostly used for diagnosis purposes. Error reasons may change over time as the
+///     framework evolves.
+///
+/// >TODO: determine what kind of stability guarantees we give about reasons/associated module.
+module DiemErrors {
+    /// A function to create an error from from a category and a reason.
+    fun make(category: u8, reason: u64): u64 {
```

Comment:
Should this reuse `Errors::make`, or is there some value in keeping these totally independent?

Reply:
Good question. I think making the base `Errors::make` public and using that makes sense here. I'll update.